### PR TITLE
Enhance artifactId parsing to perform string interpolation substitution

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: api/app/lib/SimpleScalaParser.scala [hunk 1]
```diff
@@ -68,14 +68,32 @@ trait SimpleScalaParser {
...
           isCrossBuilt = crossBuilt.length > 1
         )
       }
     }.distinct.sortBy { l => (l.groupId, l.artifactId, l.version) }.toList
   }
 
+  private def replaceVariables(str: String, lines: Seq[String]): String = {
+    val StrInterpol1Rx = """([\w.-]*)\$([\w]+)([\$\{\}\w.-]*)""".r
+    val StrInterpol2Rx = """([\w.-]*)\$\{([\w]+)\}([\$\{\}\w.-]*)""".r
+    val ValRx = """\s*[\w]+\s*([\w]+)\s*=\s*"([^"]*)"\s*""".r
```
Comment (benwaffle) [inline api/app/lib/SimpleScalaParser.scala]: couldn't you have used the `interpolate` function?

Comment (gheine) [reply api/app/lib/SimpleScalaParser.scala]: Not sure what that does?

Comment (benwaffle) [reply api/app/lib/SimpleScalaParser.scala]: substitute variables for their values. It's on line 44 in this file:
```scala
  /**
    * This method will substitute any variables for their values. For
    * literals, we strip the quotes.
    */
  def interpolate(value: String): String = {
    variables.find(_.name == value) match {
      case None => SimpleScalaParserUtil.cleanVariable(value)
      case Some(variable) => variable.value
    }
  }
```

Comment (benwaffle) [reply api/app/lib/SimpleScalaParser.scala]: SimpleScalaParser already parses variables out of this file, but it looks like you're doing the same work in your regex.

```scala
38: lazy val variables: Seq[SimpleScalaParserUtil.Variable] = SimpleScalaParserUtil.parseVariables(lines)
```

**Pull request merged**